### PR TITLE
[Logs]: Changed the logic to stop the decoder

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Changed the logic to stop the decoder.

### Motivation

Fix race conditions and tests from panicking.

### CODE REVIEW

File: pkg/logs/input/tailer/tailer_nix.go [hunk 1]
```diff
@@ -62,8 +62,8 @@ func (t *Tailer) readForever() {
 				t.wait()
 				continue
 			}
-			t.decoder.InputChan <- decoder.NewInput(inBuf[:n])
 			t.incrementReadOffset(n)
+			t.decoder.InputChan <- decoder.NewInput(inBuf[:n])
```
Comment (tmichelet) [inline pkg/logs/input/tailer/tailer_nix.go]: why are we doing this change? I find it dangerous to increment the read offset before we passed the data around

File: pkg/logs/decoder/line_handler.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func (h *SingleLineHandler) run() {
 	for line := range h.lineChan {
 		h.process(line)
 	}
-	h.outputChan <- newStopOutput()
+	close(h.outputChan)
```
Comment (tmichelet) [inline pkg/logs/decoder/line_handler.go]: go noob question: when we close a channel, it first delivers all its content before actually closing, right?

Comment (ajacquemot) [reply pkg/logs/decoder/line_handler.go]: this is correct

File: pkg/logs/input/tailer/tailer_nix.go (hunk 1 above)
Comment (ajacquemot) [reply pkg/logs/input/tailer/tailer_nix.go]: I thought this would fix some bugs, I'll revert.

**Review by tmichelet — APPROVED** [review]

**Pull request merged**